src/monattack.cpp

```diff
@@ -4647,3 +4648,19 @@ bool mattack::dodge_check( monster *z, Creature *target ) {
     }
     return false;
 }
+
+void mattack::lay_egg( monster *z )
+{
+    monster &critter = *z;
+    const furn_t &furn = g->m.furn( critter.pos() ).obj();
+
+    if( !critter.is_dead() && critter.has_flag( MF_EGGLAYING ) && furn.has_flag( TFLAG_NEST ) &&
```

Comment:
@Coolthulhu I don't have any idea what you mean by `MF_SEEKS_NEST`, and I can't find a way to make a `lay_egg` attack.

I barely understand how the whole attack thing works and there isn't much documentation about it.
From what I believe the `for( const auto &sp_type : type->special_attacks )` loop in move() method in monmove.cpp tries somehow to get attack type of that monster, and then call it.

So I made `layegg()` into a bool and added `"special_attacks": [ [ "LAY_EGG" ] ],` to mon_chicken, but the game throws me that is an invalid special_attacks.

Defined `flag_map["LAY_EGG"] = MF_LAY_EGG;` in monstergenerator.cpp and added `MF_LAY_EGG` in mtype.h.

What else do I have to do to create a new special attack?